bcda/api/requests.go

```diff
@@ -517,20 +524,20 @@ func (h *Handler) bulkRequest(w http.ResponseWriter, r *http.Request, reqType se
 
 	newJob.ID, err = rtx.CreateJob(ctx, newJob)
 	if err != nil {
-		log.API.Error(err)
+		logger.Error(err)
 		h.RespWriter.Exception(w, http.StatusInternalServerError, responseutils.DbErr, "")
 		return
 	}
 
 	if newJob.ID != 0 {
-		requestID := gcmw.GetReqID(r.Context())
-		log.API.Infof("requestID %s jobID %d", requestID, newJob.ID)
+		logging.LogEntrySetField(ctx, "job_id", newJob.ID)
+		logger = logging.GetLogEntry(r.Context())
```

Comment:
I wonder if `LogEntrySetField` should return the logger instead of, or in addition to, the new context? seems like it would simplify lines like this.

```go
logger = logging.LogEntrySetField(...)
```

or 

```go
logger, _ = logging.LogEntrySetField(...)
```

I think the other confusing thing is that `r.Context()` here would be the _old_ context, right? Since `LogEntrySetField` returns the new context 🤔 So in theory if the request context didn't already have the log key, I think there wouldn't be logger here when you look for it from `r.Context()`.

Reply:
I left the `GetLogEntry` and `LogEntrySetField` (whose name I will be updating to `SetLogEntry`) separate since these functions will be used in the worker, too. There may be cases where we just want to retrieve the logger without updating the fields at the same time. 

Maybe we leave the `GetLogEntry` and update `LogEntrySetField`  to additionally return the logger when it gets updated?

That is also my mistake on using the old context there -- I'll fix that line right now!

Reply:
OK, I updated the names to be more clear and I also realized that we are utilizing one of Chis middleware functions that would have had some conflict (GetLogEntry). 

I didn't update the return type yet -- let me know what your thoughts are on my last comment! :)

Reply:
Yeah I think we can leave (what used to be) GetLogEntry alone, and update the SetEntry method to return the logger. Thanks!

Reply:
changes have been pushed!